nautobot/utilities/drf_views.py

```diff
@@ -0,0 +1,663 @@
+        elif "export" in request.GET and hasattr(model, "to_yaml"):
+            response = HttpResponse(self.queryset_to_yaml(), content_type="text/yaml")
+            filename = f"nautobot_{self.queryset.model._meta.verbose_name_plural}.yaml"
+            response["Content-Disposition"] = f'attachment; filename="{filename}"'
+            return response
+
+        # Fall back to built-in CSV formatting if export requested but no template specified
+        elif "export" in request.GET and hasattr(model, "to_csv"):
+            response = HttpResponse(self.queryset_to_csv(), content_type="text/csv")
+            filename = f"nautobot_{self.queryset.model._meta.verbose_name_plural}.csv"
+            response["Content-Disposition"] = f'attachment; filename="{filename}"'
+            return response
+
+    def queryset_to_yaml(self):
+        """
+        Export the queryset of objects as concatenated YAML documents.
+        """
+        yaml_data = [obj.to_yaml() for obj in self.queryset]
+
+        return "---\n".join(yaml_data)
+
+    def queryset_to_csv(self):
+        """
+        Export the queryset of objects as comma-separated value (CSV), using the model's to_csv() method.
+        """
+        csv_data = []
+        custom_fields = []
+
+        # Start with the column headers
+        headers = self.queryset.model.csv_headers.copy()
+
+        # Add custom field headers, if any
+        if hasattr(self.queryset.model, "_custom_field_data"):
+            for custom_field in CustomField.objects.get_for_model(self.queryset.model):
+                headers.append(custom_field.name)
+                custom_fields.append(custom_field.name)
+
+        csv_data.append(",".join(headers))
+
+        # Iterate through the queryset appending each object
+        for obj in self.queryset:
+            data = obj.to_csv()
+
+            for custom_field in custom_fields:
+                data += (obj.cf.get(custom_field, ""),)
+
+            csv_data.append(csv_format(data))
+
+        return "\n".join(csv_data)
+
+    def list(self, request, *args, **kwargs):
+        context = {}
+        if "export" in request.GET:
+            model = self.queryset.model
+            content_type = ContentType.objects.get_for_model(model)
+            return self.check_for_export(request, model, content_type)
+        return Response(context)
+
+
+class ObjectDeleteViewMixin(NautobotViewSetMixin, mixins.DestroyModelMixin):
+    logger = logging.getLogger("nautobot.views.ObjectDeleteView")
+
+    def destroy(self, request, *args, **kwargs):
+        context = {}
+        if request.method == "POST":
+            return self.perform_destroy(request, **kwargs)
+        return Response(context)
+
+    def perform_destroy(self, request, **kwargs):
+        self.obj = self.get_object()
+        form = ConfirmationForm(request.POST)
+        if form.is_valid():
+            return self.form_valid(form)
+        else:
+            return self.form_invalid(form)
```

Comment:
It's not obvious to me why `destroy` and `perform_destroy` are separate methods? They seem short enough that they could easily be consolidated. Is this to facilitate overrides or something?

Same question for `create_or_update`/`perform_create_or_update`, `bulk_destroy`/`perform_bulk_destroy`, etc.

Reply:
Yeah I think Im trying to emulate the workflow of the original mixins but by using forms instead of serializers.
https://github.com/encode/django-rest-framework/blob/master/rest_framework/mixins.py

Reply:
It is to facilitate overrides where the core machinery doesn't need to always be overloaded like our `ObjectDetailView.post()` method today.